analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverloadOptionalParameter.cs

```diff
@@ -29,83 +29,70 @@ public sealed class MethodOverloadOptionalParameter : SonarDiagnosticAnalyzer
-                        {
-                            continue;
-                        }
-                        var defaultCanBeUsed = IsMoreParameterAvailableInConflicting(hidingInfo) || !MethodsUsingSameParameterNames(hidingInfo);
-                        var isOtherFile = syntax.SyntaxTree.FilePath != hidingMethodSyntax.SyntaxTree.FilePath;
-
-                        c.ReportIssue(Diagnostic.Create(rule, syntax.GetLocation(),
-                            hidingMethodSyntax.GetLocation().GetLineSpan().StartLinePosition.Line + 1,
-                            isOtherFile
-                                ? $" in file '{new FileInfo(hidingMethodSyntax.SyntaxTree.FilePath).Name}'"
-                                : string.Empty,
-                            defaultCanBeUsed
-                                ? "can only be used with named arguments"
-                                : "can't be used"));
+                        ReportIssue(c, info);
                     }
                 },
                 SymbolKind.Method);
+
+        private static void ReportIssue(SonarSymbolReportingContext c, ParameterHidingMethodInfo hidingInfo)
+        {
+            var syntax = hidingInfo.ParameterToReportOn.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax();
+            if (syntax == null || hidingInfo.HidingMethod.ImplementationSyntax() is not { } hidingMethodSyntax)
+            {
+                return;
+            }
+
+            var defaultCanBeUsed = IsMoreParameterAvailableInConflicting(hidingInfo) || !MethodsUsingSameParameterNames(hidingInfo);
+            var isOtherFile = syntax.SyntaxTree.FilePath != hidingMethodSyntax.SyntaxTree.FilePath;
+
+            c.ReportIssue(Diagnostic.Create(Rule, syntax.GetLocation(),
+                hidingMethodSyntax.GetLocation().GetMappedLineSpan().StartLinePosition.Line + 1,
+                isOtherFile
+                    ? $" in file '{new FileInfo(hidingMethodSyntax.SyntaxTree.FilePath).Name}'"
+                    : string.Empty,
+                defaultCanBeUsed
+                    ? "can only be used with named arguments"
+                    : "can't be used"));
         }
 
+        private static List<ParameterHidingMethodInfo> GetParameterHidingInfo(IMethodSymbol methodSymbol) =>
+            methodSymbol.ContainingType
+                        .GetMembers(methodSymbol.Name)
+                        .OfType<IMethodSymbol>()
+                        .Where(m => m.TypeParameters.Length == methodSymbol.TypeParameters.Length)
+                        .Where(m => m.Parameters.Length < methodSymbol.Parameters.Length)
+                        .Where(m => !m.Parameters.Any(p => p.IsParams))
+                        .Where(candidateHidingMethod => IsMethodHidingOriginal(candidateHidingMethod, methodSymbol))
+                        .Where(candidateHidingMethod => methodSymbol.Parameters[candidateHidingMethod.Parameters.Length].IsOptional)
```

Comment:
Opinionated: what about:
```suggestion
                        .Where(m => m.TypeParameters.Length == methodSymbol.TypeParameters.Length
                              && m.Parameters.Length < methodSymbol.Parameters.Length
                              && !m.Parameters.Any(p => p.IsParams)
                              && IsMethodHidingOriginal(m, methodSymbol)
                              && methodSymbol.Parameters[m.Parameters.Length].IsOptional)
```
It's a bit more readable to me, totally optional

Reply:
I would not change it now. Mainly because I want to merge it before the next peachee run.